Updated the equalizer plugin to use sample exact controls

curlymorphic opened the pull request:
Updated the equalizer plugin to use sample exact controls for the frequency res and bandwidth controls.
For this i had to do some major refactoring within the processAudioBuffer()
function. Originally each filter processed the buffer in a separate loop, This has been improved and each filter is now processed per frame.

Sti2nd commented:
:+1:

tresf commented:
@curlymorphic, do you need review on this prior to merging?  Since EQ is something you created, I'm inclined to merge any PRs you send against it, but I also know you and Vesa were discussing this quite a bit, so I don't want to prematurely merge.

curlymorphic commented:
> if you're hoping for feedback first.

I fell it may be better if @diizy give's this the once over, I have followed the guidelines for including sample exactness but not had any code doing so "passed" yet.

diizy commented on plugins/Eq/EqFilter.h:
```diff
@@ -136,6 +136,12 @@ class EqFilter : public StereoBiQuad
 		}
 	}
 
+	virtual void tick( sampleFrame *frame )
```
What is the purpose of this function? Just seems like adding extra overhead... would probably be better to use inline functions for things such as this

curlymorphic commented:
Removed the tick() function, and replaced the invoking code with
direct calls to update().

fixed a bug, where the low shelf bandwidth was getting the incorrect data.
